esmvalcore/cmor/table.py

```diff
@@ -98,22 +92,26 @@ def _load_table(self, json_file):
                 return
             table = TableInfo()
             header = raw_data['Header']
-            table.name = header['table_id'][6:].split('_')[-1]
+            table.name = header['table_id'].split(' ')[-1]
             self.tables[table.name] = table
 
             generic_levels = header['generic_levels'].split()
             table.frequency = header.get('frequency', '')
-            table.realm = header.get('realm', '')
+            self.var_to_freq[table.name] = {}
 
             for var_name, var_data in raw_data['variable_entry'].items():
                 var = VariableInfo('CMIP6', var_name)
-                if 'frequency' in var_data:
-                    var.frequency = var_data['frequency']
-                else:
-                    var.frequency = table.frequency
-                var.read_json(var_data)
+                var.read_json(var_data, table.frequency)
                 self._assign_dimensions(var, generic_levels)
                 table[var_name] = var
+                self.var_to_freq[table.name][var_name] = var.frequency
+
+            if not table.frequency:
```

Comment:
Probably this should be `table.frequency is ''`.

Reply:
no, the check on the existence of `frequency` is needed here since it's either a valid value of `None` (from what I remember)

Reply:
Sorry, @valeriupredoi, I don't understand. This is not a test for existence. It checks the truthiness of `table.frequency`. What do you mean? A valid value of `None` or what?

Reply:
@valeriupredoi care to answer and/or resolve (in the github sense) this conversation?